/**********************************************************************************************************************
 *  GENIS KOREA SOFTWARE
 *  -------------------------------------------------------------------------------------------------------------------
 *  Copyright (c) 2019-2021 GENIS KOREA Inc.                                                 All rights reserved.
 *
 *                This software is copyright protected and proprietary to GENIS KOREA.
 *                Restricted Distribution:
 *                  Not to be distributed This source code to anyone without the approval from GENIS KOREA.
 *
 *  -------------------------------------------------------------------------------------------------------------------
 *  DESCRIPTION
 *  -------------------------------------------------------------------------------------------------------------------
 *  Filename: Csm_Dbcs.h
 *  Created: Apr 3, 2024
 *  Author: sbkang
 *
 *********************************************************************************************************************/

#ifndef CHARGER_DBC_CSM_DBCS_H_
#define CHARGER_DBC_CSM_DBCS_H_
 
/**********************************************************************************************************************
 *  INCLUDES
 *********************************************************************************************************************/
#include "csm_secc.h"
#include "csm_charger.h"

/**********************************************************************************************************************
 *  GLOBAL CONSTANT MACROS
 *********************************************************************************************************************/

/**********************************************************************************************************************
 *  GLOBAL FUNCTION MACROS
 *********************************************************************************************************************/

/**********************************************************************************************************************
 *  GLOBAL DATA TYPES AND STRUCTURES
 *********************************************************************************************************************/
//#pragma pack(push,1)
//#pragma pack(pop)

/**********************************************************************************************************************
 *  GLOBAL DATA PROTOTYPES
 *********************************************************************************************************************/

/**********************************************************************************************************************
 *  GLOBAL FUNCTION PROTOTYPES
 *********************************************************************************************************************/
 
#ifdef __cplusplus
extern "C" {
#endif

void can_csm_secc_from_map(can_csm_secc_t *secc_bus, uint16 *in);
uint8 can_csm_secc_to_map(const can_csm_secc_t *secc_bus, uint16 *out);

void can_csm_charger_from_map(can_csm_charger_t *charger_bus, uint16 *in);
uint8 can_csm_charger_to_map(const can_csm_charger_t *charger_bus, uint16 *out);

#ifdef __cplusplus
}
#endif

#endif /* CHARGER_DBC_CSM_DBCS_H_ */
